### ISSUE TITLE
[Feature request] Function to get where a symbol is defined

### ISSUE THREAD

**Issue by Rangi42** [gbdev/rgbds#910]
IDEs make it easy to find where symbols are defined, but rgbasm doesn't have one. It's also not reliable to grep for "`symbol:`" or even "`symbol\s*:`" because it might be defined as `MACRO symbol`, `symbol /*comment*/ :`, `def symbol equ`, etc. Furthermore, grep may turn up definitions in unrelated files.

A `STRDEF` function would be convenient to get the filename and line number where symbol is defined.

For example, test.asm:

```
SECTION "test", ROM0
N = 42
Label:
println STRDEF(Label), ":Label"  ; test.asm:3:Label
println "N is defined at ", STRDEF(N)  ; N is defined at test.asm:1
assert !STRCMP(STRDEF(nonexistent), "")
purge N
DEF N EQUS "string"
macro find
    println "'\1' is at ", STRDEF(\1)
endm
    find N  ; 'N' is at test.asm:8
    find find  ; 'find' is at test.asm:9
```

`STRDEF(nonexistent)` could also have a `-Wextra` warning that the symbol is not defined.

**Comment by ISSOtm** [comment]
This is very, very shakey imo... symbols are defined as part of a context stack plus an `EQUS` stack, and the latter is especially troublesome because without `DEF`, a definition may be surprising.

What are the intended use cases? There may be other ways to fulfill them.

**Comment by Rangi42** [comment]
The use case is to find where a symbol is defined.

One way this would help is if you're generating symbols via some macro or rept block, and it's not taking the value you expect.

Another would be to learn more about a project. This is of limited use if there are a lot of object files, but if (for example) there's one `main.asm` that includes most of the code, you can put `println STRDEF(foo)` at the end to see where `foo` was defined, no matter whether it's a label, constant, or macro.

**Comment by ISSOtm** [comment]
Why do you need to find where a symbol is defined? If it's for printing, we could define a `printdef` or `printloc` keyword that prints the definition stack to standard output, for example.

**Comment by ISSOtm** [comment]
(Why Ctrl+Enter suddenly decided to target "Close with comment" is beyond me.)

**Comment by Rangi42** [comment]
Users often ask "what is (some symbol)?", and it's not always sufficient to tell them "`git grep 'symbol:'`" (e.g. `git grep wMon5` won't find `for N, 1, 6+1 \ wMon{d:N}::`).

I briefly considered a `printdef` or `backtrace` statement, but it's just as easy to type `println strdef`, and allows you to use the string for something besides immediately printing. (At the very least, it could be part of a formatted message like "`lb is defined at macros/code.asm:42`". Or maybe you'd want to replace the "` -> `"s with newlines, or print only the first or last location in a chain of `->`s.)

**Comment by aaaaaa123456789** [comment]
If you're going for the "reusable" form, then split it into file and line functions — that way they can be combined however you want.

**Comment by ISSOtm** [comment]
Then this should happen after #67.

**Comment by Rangi42** [comment]
In that case I'll just close this for now.

I overlooked that you can already find where a symbol is defined by trying to redefine it:

```diff
     maskbits NUM_DEX_ENTRY_BANKS
+maskbits equ 42
```
```
ERROR: main.asm(43) -> engine/items/item_effects.asm(774):
    'maskbits' already defined at main.asm(1) -> constants.asm(3) -> macros.asm(6) -> macros/code.asm(26)
```
